internal/pipeline/types.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// This file defines the type structures used by pipeline
+package pipeline
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Function defines an item in the pipeline function list
```

Comment:
nit: I would recommend changing the order of the structs (top-down) to improve the readability. Start with the top level struct i.e. Pipeline, then Function ...